lib/hanami/extensions/view/slice_configured_part.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Hanami
+  module Extensions
+    module View
+      # Provides slice-specific configuration and behavior for any view part class defined within a
+      # slice's module namespace.
+      #
+      # @api private
+      # @since 2.1.0
+      class SliceConfiguredPart < Module
+        attr_reader :slice
+
+        # @api private
+        # @since 2.1.0
+        def initialize(slice)
+          super()
+          @slice = slice
+        end
+
+        # @api private
+        # @since 2.1.0
+        def extended(klass)
+          define_new
+        end
+
+        # @api private
+        # @since 2.1.0
+        def inspect
+          "#<#{self.class.name}[#{slice.name}]>"
+        end
+
+        private
+
+        # Defines a `.new` method on the part class that provides a default `rendering:` argument of
+        # a rendering coming from a view configured for the slice. This means that any part can be
+        # initialized standalone (with a `value:` only) and still have access to all the integrated
+        # view facilities from the slice, such as helpers. This is helpful when unit testing parts.
+        #
+        # @example
+        #   module MyApp::Views::Parts
+        #     class Post < MyApp::View::Part
+        #       def title_tag
+        #         helpers.h1(value.title)
+        #       end
+        #     end
+        #   end
+        #
+        #   # Useful when unit testing parts
+        #   part = MyApp::Views::Parts::Post.new(value: hello_world_post)
+        #   part.title_tag # => "<h1>Hello world</h1>"
+        #
+        # @api private
+        # @since 2.1.0
+        def define_new
+          slice = self.slice
+
+          define_method(:new) do |**args|
+            return super(**args) if args.key?(:rendering)
+
+            slice_rendering = Class.new(Hanami::View)
+              .configure_for_slice(slice)
+              .new
+              .rendering
```

Comment:
For the sake of efficiency, I originally had this rendering initialised _outside_ the `defined_method` block and then accessed within it via standard closure behaviour, but this turned out to cause some infinite loops since `Hanami::View.configure_for_slice` creates a part class, which would bring us here, which would. then call `Hanami::View.configure_for_slice` again, and so on.

Creating the rendering inside the method still feels reasonable enough given this code will only ever be exercised in testing. Parts created as part of ordinary view rendering will always have an explicit `rendering:` argument provided.